docs/semgrep-secrets/glossary.md

```diff
@@ -0,0 +1,90 @@
+<i class="fas fa-external-link fa-xs"></i> Secrets-specific ruleset(https://semgrep.dev/p/secrets).
+
+### Sources of rules
+
+The Semgrep Registry contains rules imported from various repositories, including non-Semgrep individuals or groups, such as Trail of Bits and GitLab. You can view a rule's `license` key to ensure the license meets your needs.
+
+## Revocation
+
+Revoking a secret makes it inactive. This can be done when a secret isn't required anymore or if a secret becomes compromised.
+
+## Rotation
+
+Rotating secrets is the process of updating a secret regularly. If a secret is leaked, regular rotation can ensure that the credential is valid only for a limited time. Rotating secrets can also minimize risk due to the reuse of secrets.
+
+## Ruleset
+
+Rulesets are rules related through a programming language, OWASP category, or framework. Rulesets are curated by the team at Semgrep and updated as new rules are added to the Semgrep Registry.
+
+## Scan target
+
+A scan target is any file or collection of files and directories that Semgrep can scan. While Semgrep can scan **any** text file through `generic` mode, Semgrep primarily scans the following:
+
+### Codebase
+
+Any code files within a specified directory and its subdirectories.
+
+### Project
+
+A repository or codebase that you have added to Semgrep Cloud Platform for scanning along with finding metadata and other Semgrep data and resources.
+
+### Repository
+
+A location, typically remote, for source code, including metadata relating to the source code. Semgrep supports Git repositories.
+
+## Secret
+
+Secrets are pieces of sensitive information crucial for securing applications and their data. This information can include API keys, access credentials, SSH keys, certificates, and more. If secrets are stored in source code, they can be "leaked," allowing internal and external malicious actors to use this information for unauthorized access.
+
+## Semantic analysis
+
+Semantic analysis refers to Semgrep Secrets' ability to understand how data is used in your code. Semgrep Secrets uses several mechanisms to perform semantic analysis, including [<i class="fa-regular fa-file-lines"></i> data-flow analysis](/writing-rules/data-flow/data-flow-overview) and [<i class="fa-regular fa-file-lines"></i> constant propagation](/writing-rules/data-flow/constant-propagation), allowing Secrets to track data, such as variables, and the flow of that data across files and functions in your codebase.
+
+## Validation state
+
+The validation state of a secret provides information on whether a secret, if leaked, poses an immediate security threat. Current Semgrep validation states for a secret include:
+
+- **Confirmed valid:** Semgrep made an HTTP request using the secret, and it returned an HTTP status code of 200 or similar **and** some indication of valid access. For example, a service can include a `"message": "ok"` in the response body.
```

Comment:
Wonder if we should turn this into a component and then import it onto the pages that need it (not something we have to do for this PR unless you want to).